### PR TITLE
NICPS-411: Added Email To SMS Code

### PR DESCRIPTION
**Pull request by deepakzimb**
Added code to send Email as SMS.
Testing Instructions added in ticket to verify changes.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	private static final String smsDomain = "esms.gov.in";
+	private static final String smsApiUrl = getsmsapiurl();
+	private static final String smsUsername = "zimbra.sms";
+	private static final String smsPin = "moorxu81";
+	private static final String smsSenderId = "NICSMS";
+
+	private EmailToSMS() {
+	}
+
+	static {
```
Comment (zimsuchitgupta) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: Please remove this block. As you already registered in ZimbraLmtpBackend.java

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		try {
+			sendEmailSMS(pm, account);
+		} catch (ServiceException e) {
+			ZimbraLog.mailbox.error("Failed to send SMS forwardWithoutDelivery ServiceException  ", e);
+		}
+	}
+
+	private void sendEmailSMS(ParsedMessage pm, Account act) throws ServiceException {
+		String recipients = pm.getRecipients();
+		String[] splitedRecipients = recipients.split(",");
```
Comment (jkester1986) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]:
```suggestion
		String[] splitRecipients = recipients.split(",");
```
`splited` is incorrect word, `split` is both an action (to split), and past-tense (something that has been broken up)

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		String subject = pm.getSubject();
+		String message = pm.getFragment(act.getLocale());
+		String fullMessage = encode(subject + "\n" + message);
+
+		for (int i = 0; i < splitedRecipients.length; i++) {
+			String rcptAddresses = splitedRecipients[i];
+			String[] splitedMobNumber = rcptAddresses.indexOf("<") < 0 ? rcptAddresses.split("@")
```
Comment (jkester1986) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: same here, `splited` should be `split`

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+			String mobNumber = splitedMobNumber[0];
+			String mobNumberDomain = splitedMobNumber[1];
+
+			if (mobNumberDomain.toLowerCase().equals(smsDomain)) {
+				call(fullMessage, mobNumber);
+			}
+		}
+	}
+
+	private void call(String message, String mnumber) throws ServiceException {
```
Comment (jkester1986) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: we aren't calling the number, just texting it, so better function name would be `text`

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+		String smsApiUrl = Provisioning.getInstance().getConfig().getAttr(Provisioning.A_zimbraSMSApiUrl);;
+		String smsUsername = Provisioning.getInstance().getConfig().getAttr(Provisioning.A_zimbraSMSApiUsername);;
+		String smsPin = Provisioning.getInstance().getConfig().getAttr(Provisioning.A_zimbraSMSApiPin);;
+		String smsSenderId = Provisioning.getInstance().getConfig().getAttr(Provisioning.A_zimbraSMSApiSenderId);;
+
```
Comment (jkester1986) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: seems like this is an extra line, unless you are separating the below var from the rest?

**Pull request merged**